### ISSUE TITLE
Clair API v1 POST layer seems to be failing with certain (empty) layers. 

### ISSUE THREAD

**Issue by Sudneo** [quay/clair#684]
tl;dr POST-ing a layer to clair with API v1 seems to generally work. Sometimes though, with empty layers, the POST fails, even if a parent for those is specified.

### Description of Problem / Feature Request

POST-ing layers to clair with API v1 might sometimes fail if the layer is empty. I read on another issue (https://github.com/coreos/clair/issues/268) that this happens when clair does not see any OS/Package manager in the layer, and to solve it is necessary to set a parent layer that clair will recognize and use as a base. 

I try posting two layers separately:
```

{
  "Layer": {
    "Format": "Docker",
    "Name": "sha256:33b176c904dea23b745303a0a69511750d009ac04fd9957520975a0546a67799",
    "ParentName": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
    "Path": "http:///192.168.200.100:5000/v2/nginx/blobs/sha256:33b176c904dea23b745303a0a69511750d009ac04fd9957520975a0546a67799"
  }
}
```

Which returns 200 OK, and then the lower layer:

```
{
  "Layer": {
    "Format": "Docker",
    "Name": "sha256:1ba02017c4b278fe664e669af5e2eff59f92259df6c2f15261f150ae66599dff",
    "ParentName": "sha256:33b176c904dea23b745303a0a69511750d009ac04fd9957520975a0546a67799",
    "Path": "http:///192.168.200.100:5000/v2/nginx/blobs/sha256:1ba02017c4b278fe664e669af5e2eff59f92259df6c2f15261f150ae66599dff"
  }
}
```

### Expected Outcome

Since I specified a parent layer, I would assume that Clair would work and return 200 OK.

### Actual Outcome

I get 402 with error:

> failed because OS/package manager is not supported

I can see from the logs:
```
{"Event":"processing layer","Level":"debug","Location":"worker.go:73","Time":"2019-01-04 15:19:44.315714","engine version":3,"format":"Docker","layer":"sha256:1ba02017c4b278fe664e669af5e2eff59f92259df6c2f15261f150ae66599dff","parent layer":"sha256:33b176c904dea23b745303a0a69511750d009ac04fd9957520975a0546a67799","path":"http://192.168.200.100:5000/v2/nginx/blobs/sha256:1ba02017c4b278fe664e669af5e2eff59f92259df6c2f15261f150ae66599dff"}
{"Event":"Namespace unknown","Level":"warning","Location":"worker.go:211","Time":"2019-01-04 15:19:45.367745","feature name":"attr","feature version":"1:2.4.47-2","layer":"sha256:1ba02017c4b278fe664e669af5e2eff59f92259df6c2f15261f150ae66599dff"}
{"Event":"Handled HTTP request","Level":"info","Location":"router.go:57","Time":"2019-01-04 15:19:45.368085","elapsed time":1052417608,"method":"POST","remote addr":"192.168.200.1:35922","request uri":"/v1/layers","status":"422"}
```

This might not be a problem in all cases, but it is a problem when the failed layer is the upper or is the one on which the upper relies as a parent (at least in my experience, this completely broke the scan).

### Additional Information

I see that clair:v2.0.7 image is relatively new, but it does not support API v3 (? Every post to v3 URLs returns 404).

### Environment

- Clair version/image: clair:v2.0.7
- Clair client name/version: curl/Python3.5 Requests
- Host OS: Ubuntu
- Kernel (e.g. `uname -a`): 4.15.0-43-generic #46-Ubuntu SMP
- Kubernetes version (use `kubectl version`): N/A
- Helm version (use `helm version`): N/A
- Network/Firewall setup: Virtualbox host-only network.

**Comment by jzelinskie** [comment]
It cannot be just any parent. Because layers are just filesystem diffs, you need a base layer to start with. When I say base layer, I mean a layer that added the filesystem to the image. It looks like 33b176 isn't one of these or is adding an unsupported one.

You also need unique names and not just the SHA256. I recommend concatenating the SHA with some identifier for the image. This is something we've fixed in v3.
